src/messaging/ExchangeContext.cpp

```diff
@@ -322,8 +323,11 @@ ExchangeContext::ExchangeContext(ExchangeManager * em, uint16_t ExchangeId, cons
 
     SetAckPending(false);
 
-    // Do not request Ack for multicast
-    SetAutoRequestAck(!session->IsGroupSession());
+    if (session->AllowsMRP())
+    {
+        // Do not request Ack for multicast
+        SetAutoRequestAck(!session->IsGroupSession());
```

Comment:
This makes no sense now.  If AllowsMRP(), then IsGroupSession() is definitely false, right?

Fundamentally, we seem to be doing this thing where:

1. We `SetAutoRequestAck(!session->AllowsMRP())` here.
2. When we go to actually maybe request acks, we check `session->AllowsMRP()` again just in case someone did `SetAutoRequestAck(true)` on us in the meantime, I guess?  But then why do we bother checking the session for the initial value of "auto request ack"?  I guess it's OK, but seems to be confusing to both whoever wrote this code initially and you.

Reply:
It is indeed a bit confusing in how all these separately named checks actually are directly mappable onto each other and so redundant.
But why `SetAutoRequestAck(!session->AllowsMRP());` ?
We actually need to do `SetAutoRequestAck(session->AllowsMRP());`. Did you mean that?

Reply:
>  But why SetAutoRequestAck(!session->AllowsMRP()); ?

Sorry, the `!` should not have been there.  It was late in the day....  

> I guess you meant that?

Yes. @pidarped

Reply:
It is now `SetAutoRequestAck(session->AllowsMRP());`